c_src/erl_optics.c

```diff
@@ -511,13 +548,17 @@ static ERL_NIF_TERM eo_optics_poll(
     if (!optics) return ERROR("get_optics");
     if (!poller) return ERROR("poller_allocation_error");
 
-    ctx->env = env;
-    ctx->map = enif_make_new_map(env);
-
+    if (erlang_poller != 0){
```

Comment:
Would `if (erlang_poller)` work here? Do you think it's a good idea?

Reply:
Yes, but I figured that since I use if (poller) to check if poller is null, writing this part like this made it more obvious that it's a boolean check and not an allocation check.